network/src/counters.rs

```diff
@@ -171,6 +171,33 @@ pub fn rpc_bytes(
     ])
 }
 
+pub static INVALID_RPC_MESSAGES: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "libra_network_invalid_rpc_messages",
+        "Number of invalid RPC messages",
+        &["role_type", "network_id", "peer_id"]
+    )
+    .unwrap()
+});
+
+pub static INVALID_DIRECT_SEND_MESSAGES: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "libra_network_invalid_ds_messages",
+        "Number of invalid direct send messages",
+        &["role_type", "network_id", "peer_id"]
+    )
+    .unwrap()
+});
+
+pub static INVALID_PEER_SEND_FAILURES: Lazy<IntCounterVec> = Lazy::new(|| {
```

Comment:
seems somewhat higher level than the above, why not include protocol_id in the previous two? or is this lower level (closer to the wire)

Reply:
Oops, this actually doesn't look like it's used.

Reply:
Fixed